pkg/meta/model/table.go

```diff
@@ -867,6 +872,69 @@ func (pi *PartitionInfo) SetOriginalPartitionIDs() {
 	pi.OriginalPartitionIDsOrder = ids
 }
 
+// IDsInDDLToIgnore returns a list of IDs that the current
+// session should not see (may be duplicate errors on insert/update though)
+// For example during truncate or drop partition.
+func (pi *PartitionInfo) IDsInDDLToIgnore() []int64 {
+	// TODO:
```

Comment:
There are multiple PRs/enhancements that will build upon this, especially for DROP and TRUNCATE partition.